pkg/container/ifset/set.go

```diff
@@ -0,0 +1,378 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package ifset
+
+import (
+	"iter"
+	"maps"
+)
+
+// Member is an empty interface to hold values in Set
+type Member interface {
+	String() string
+}
```

Comment:
This is just a duplicate of the [fmt.Stringer](https://pkg.go.dev/fmt#Stringer) interface.

Anyway, I think we should avoid constraining the types to be used for a Set with this interface. Instead, I'd rather use `any` for broader reusability of the data structure and leave the responsibility of stringify the elements to the user, whenever possible (e.g: for the `mapstate` package we can write a function that first assert the type and then produce a string accordingly).

Reply:
Right, added this for debugging and did not think at that point this limits the member types we can use in a set! Will change back to `any`!